[DataGridPro] Add option to infer detail panel height from content (#5163)
mui/mui-x: docs/data/data-grid/master-detail/master-detail.md

To use this feature, pass a function to the `getDetailPanelContent` prop with the content to be rendered inside the panel.
Any valid React element can be used as the row detail, even another grid.

By default, the detail panel height is 500px.

> However, this can be customized by passing a function to the `getDetailPanelHeight` prop.

Reviewer: ```suggestion
You can customize it by passing a function to the `getDetailPanelHeight` prop.
```

> This function must return a number or the `"auto"` string.

Reviewer: ```suggestion
This function must return either a number or the `"auto"` string.
```

> Returning a number means that the panel will use as height exactly the value provided.

Reviewer: ```suggestion
If it returns a number, then the panel will use that value (in pixels) for the height.
```

> With `"auto"`, the height is [derived](#infer-height-from-the-content) from the content.

Reviewer: ```suggestion
If it returns `"auto"`, then the height will be [derived](#infer-height-from-the-content) from the content.
```

```tsx
<DataGridPro
  getDetailPanelContent={({ row }) => <div>Row ID: {row.id}</div>}
  getDetailPanelHeight={({ row }) => 100} // Optional, default is 500px.
/>

// or

<DataGridPro
  getDetailPanelContent={({ row }) => <div>Row ID: {row.id}</div>}
  getDetailPanelHeight={({ row }) => 'auto'} // Height based on the content.
/>
```

:::info

> Both props are called with a [`GridRowParams`](/x/api/data-grid/grid-row-params/) object, allowing you to return a different value for each row.

Reviewer: ```suggestion
Both props are called with a [`GridRowParams`](/x/api/data-grid/grid-row-params/) object, which lets you return a different value for each row.
```

:::

> To expand a row, click on the `+` icon or press <kbd class="key">Space</kbd> inside the detail toggle column.

Reviewer: ```suggestion
To expand a row, click on the **+** icon or press <kbd class="key">Space</kbd> inside the detail toggle column.
```

:::

## Infer height from the content

Like [dynamic row height](/x/react-data-grid/rows/#dynamic-row-height), you can also derive the detail panel height from its content.
For this, pass a function to the `getDetailPanelHeight` prop returning `"auto"`, as below:

```tsx
<DataGridPro getDetailPanelHeight={() => 'auto'} />
```

> The following example demonstrantes this option in action.

Reviewer: ```suggestion
The following example demonstrates this option in action:
```
